driver/sigmoid_focal_loss_driver.hpp

```diff
@@ -0,0 +1,774 @@
+        miopenCreateTensorDescriptor(&doutputDesc);
+        miopenCreateTensorDescriptor(&dinputDesc);
+        miopenCreateTensorDescriptor(&dtargetDesc);
+
+        data_type = miopen_type<TIO>{};
+    }
+
+    std::vector<int> ComputeStrides(std::vector<int> input);
+    int AddCmdLineArgs() override;
+    int ParseCmdLineArgs(int argc, char* argv[]) override;
+    InputFlags& GetInputFlags() override { return inflags; }
+
+    int GetandSetData() override;
+
+    int AllocateBuffersAndCopy() override;
+
+    int RunForwardGPU() override;
+    int RunForwardCPU();
+
+    int RunBackwardGPU() override;
+    int RunBackwardCPU();
+
+    int VerifyBackward() override;
+    int VerifyForward() override;
+    ~SigmoidFocalLossDriver() override
+    {
+        miopenDestroyTensorDescriptor(inputDesc);
+        miopenDestroyTensorDescriptor(targetDesc);
+        miopenDestroyTensorDescriptor(outputDesc);
+        miopenDestroyTensorDescriptor(doutputDesc);
+        miopenDestroyTensorDescriptor(dinputDesc);
+        miopenDestroyTensorDescriptor(dtargetDesc);
+    }
+
+private:
+    InputFlags inflags;
+
+    miopenTensorDescriptor_t inputDesc;
+    miopenTensorDescriptor_t targetDesc;
+    miopenTensorDescriptor_t outputDesc;
+    miopenTensorDescriptor_t doutputDesc;
+    miopenTensorDescriptor_t dinputDesc;
+    miopenTensorDescriptor_t dtargetDesc;
+
+    std::unique_ptr<GPUMem> input_dev;
+    std::unique_ptr<GPUMem> target_dev;
+    std::unique_ptr<GPUMem> output_dev;
+    std::unique_ptr<GPUMem> doutput_dev;
+    std::unique_ptr<GPUMem> dinput_dev;
+    std::unique_ptr<GPUMem> dtarget_dev;
+    std::unique_ptr<GPUMem> workspace_dev;
+
+    std::vector<TIO> input;
+    std::vector<TIO> target;
+    std::vector<TIO> output;
+    std::vector<TIO> outputHost;
+    std::vector<TIO> doutput;
+    std::vector<TIO> dinput;
+    std::vector<TIO> dinputHost;
+    std::vector<TIO> dtarget;
+    std::vector<TIO> dtargetHost;
+    std::vector<TIO> workspace;
+
+    float alpha;
+    float gamma;
+    float divisor;
+    bool isContiguous;
+    bool isTargetGradientComputed;
+    miopenLossReductionMode_t reduction;
+
+    size_t workSpaceSizeInBytes;
+};
+
+template <typename TIO>
+int SigmoidFocalLossDriver<TIO>::ParseCmdLineArgs(int argc, char* argv[])
+{
+    inflags.Parse(argc, argv);
+
+    if(inflags.GetValueInt("time") == 1)
+    {
+        miopenEnableProfiling(GetHandle(), true);
+    }
+    return miopenStatusSuccess;
+}
+
+template <typename TIO>
+int SigmoidFocalLossDriver<TIO>::GetandSetData()
+{
+    auto inDims              = inflags.GetValueTensor("dim-lengths").lengths;
+    alpha        = inflags.GetValueDouble("alpha");
+    gamma        = inflags.GetValueDouble("gamma");
+    isContiguous = inflags.GetValueInt("is-contiguous") == 1 ? true : false;
+    isTargetGradientComputed = inflags.GetValueInt("target-gradient") == 1 ? true : false;
+    reduction    = static_cast<miopenLossReductionMode_t>(inflags.GetValueInt("reduction"));
```

Comment:
clang-format is required.
If you use a github hook, it will be checked every time you commit.
```
./.githooks/install
git config core.hooksPath .githooks
```

Reply:
I migrated code to a new node yesterday and forgot to setup the githooks and clang-format in IDE. I have updated it.